### DISCUSSION TITLE
URLError and more

### DISCUSSION THREAD

**Discussion by RossJL** [pyvista/pyvista#2922] [Q&A]
So I'm trying to do the 'Compute Gradients of a Field' tutorial (https://docs.pyvista.org/examples/01-filter/gradients.html?highlight=gradient%20filter) but I am running into a URLError to download the carotid. The error is as follows:

URLError: <urlopen error [WinError 10061] No connection could be made because the target machine actively refused it>

Are the URL's for those examples still good or do those need to be updated? Or is there a foreseeable problem on my end? I've already tried checking for proxy use (no good). 

I've also tried using a different .stl file instead to try and compute the gradients, but anytime I have scalars = 'gradients', that doesn't work either. I just get the following:

ERROR:root:No input array. If this dataset is part of a composite dataset check to make sure that all non-empty blocks have this array.
ERROR:root:Algorithm vtkGradientFilter(0000028573F6A090) returned failure for request: vtkInformation (000002856AD2E5B0)

**Comment by akaszynski** [comment]
This links still work, can you download any example? It might be that you can't download the example on your machine.

We might want to implement a check that verifies if we can open a connection to github and download files. If not, we need to raise a more helpful error.

**Reply by RossJL** [reply]
Yes, just tried it with a couple of other example downloads and received the same error. It might be the machine.

When it comes to using it for the different .stl file, what does it mean by 'no input array'?

**Reply by RossJL** [reply]
the .stl file has no problem plotting correctly (it is loaded locally).

**Comment by MatthewFlamm** [comment]
This is where a local source of the example data is very useful: https://docs.pyvista.org/extras/vtk_data.html

If you can provide a copy of the data locally this works very well for me.
